CRM/Case/BAO/Case.php

```diff
@@ -1212,9 +1212,20 @@ public static function getCaseActivity($caseID, &$params, $contactID, $context =
    *
    */
   public static function getRelatedContacts($caseID, $skipDetails = FALSE) {
+    $caseRoles = array();
+    if (!$skipDetails) {
+      $caseInfo = civicrm_api('Case', 'getsingle', array(
```

Comment:
can we not use civicrm_api3?

There is acceptable reason for an error here so an uncaught exception would be appropriate